gluten-core/src/main/scala/org/apache/gluten/planner/property/Conv.scala

```diff
@@ -0,0 +1,106 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.gluten.planner.property
+
+import org.apache.gluten.extension.columnar.transition.{Convention, ConventionReq, Transition}
+import org.apache.gluten.ras.{Property, PropertyDef}
+import org.apache.gluten.ras.rule.{RasRule, Shape, Shapes}
+
+import org.apache.spark.sql.execution._
+
+sealed trait Conv extends Property[SparkPlan] {
```

Comment:
It would be nice to add some description for the new API and its members.

Reply:
Thanks. I will add some then.

BTW it's a RAS component as of now (and all source files in package `org.apache.gluten.extension.columnar.enumerated` and `org.apache.gluten.planner`) that isn't used by regular code path.